githttp.go

```diff
@@ -10,6 +10,14 @@ import (
 	"strings"
 )
 
+type ResolveInfo struct {
+	Path    string
+	Request *http.Request
+}
+
+// Resolve a URL path to a repo location
+type PathResolver func(info ResolveInfo) (string, error)
```

Comment:
Comment should begin with "PathResolver ...", see https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences.